src/minidriver/minidriver.c

```diff
@@ -1888,6 +1891,12 @@ md_set_cmapfile(PCARD_DATA pCardData, struct md_file *file)
 						  cont->flags & CONTAINER_MAP_DEFAULT_CONTAINER ?
 						  " (default)" : "");
 
+					/* set flag that at least one key that uses the sign key needs PinCacheAlwaysPrompt */
+					logprintf(pCardData, 7, "key_obj->user_consent: %d\n", (int) key_obj->user_consent);
+					if (key_obj->user_consent)
+						vs->need_pin_always = 1;
+					logprintf(pCardData, 7, "vs->need_pin_always %d\n", (int) vs->need_pin_always);
```

Comment:
I guess this should print `need_pin_always` only when it is being set (inside that `if` block)?

This value can be deducted from the above `user_consent` printout anyway.

Reply:
I left the `logprintf` outside to show value of `need_pin_always` after the test.
the `need_pin_always` could be an array for each ROLE, if there is ever a third type of Sign Pin.